feat(auth): Add documentation for next step in auth (#2936)
aws-amplify/docs: docs/lib/auth/fragments/ios/signin_next_steps/20_confirm_sms_mfa.md

> If you get `confirmSignInWithSMSMFACode`, the next step is to verify the user through SMS multi-factor authentication. `AuthCodeDeliveryDetails` will contain information regarding where the code has been send. You should ask the user to provide the code that they received via SMS and invoke the `confirmSignIn` api with the SMS code:

Reviewer: ```suggestion
If the next step is `confirmSignInWithSMSMFACode`, Amplify Auth has sent the user a random code over SMS, and is waiting to find out if the user successfully received it. To handle this step, your app's UI must prompt the user to enter the code. After the user enters the code, your implementation must pass the value to Amplify Auth's `confirmSignIn` API.

Note: the signin result also includes an `AuthCodeDeliveryDetails` member. It includes additional information about the code delivery such as the phone number of the SMS recipient.
```

Reviewer 2: "such as the partial phone number of the SMS recipient" (I believe this only includes the last few digits of the number)

<amplify-block-switcher>

<amplify-block name="Listener (iOS 11+)">

```swift

> func confirmSignIn() {
>     Amplify.Auth.confirmSignIn(challengeResponse: "<confirmation code received via SMS>") { result in

Reviewer 2: Here and throughout in other code samples: you can make the usage a bit clearer:
```suggestion
func confirmSignIn(confirmationCodeFromUser: String) {
    Amplify.Auth.confirmSignIn(challengeResponse: confirmationCodeFromUser) { result in
```

Author: Updated

        switch result {
        case .success(let signInResult):

>             print("Confirm sign in succeeded. Next step: \(signInResult.nextStep)")

Reviewer 2: Here and throughout in other code samples:

```suggestion
            if signInResult.isSignedIn {
                print("Confirm sign in succeeded. The user is now signed in.")
            } else {
                print("Confirm sign in succeeded. Next step: \(signInResult.nextStep)")
                // Switch on the next step to take appropriate actions. If `signInResult.isSignedIn`
                // is true, the next step is 'done', and the user is now signed in.
            }
```

Please confirm that `done` and `isSignedIn` are the correct values.

Author: Updated

        case .failure(let error):
            print("Confirm sign in failed \(error)")
        }
    }
}
```

</amplify-block>

<amplify-block name="Combine (iOS 13+)">

```swift
func confirmSignIn() -> AnyCancellable {
    Amplify.Auth.confirmSignIn(challengeResponse: "<confirmation code received via SMS>")
        .resultPublisher
        .sink {
            if case let .failure(authError) = $0 {
                print("Confirm sign in failed \(authError)")
            }
        }
        receiveValue: { signInResult in

>             print("Confirm sign in succeeded. Next step: \(signInResult.nextStep)")

Reviewer 2: Same notes as above re: function argument & success block handling

Author: Updated
